### PR TITLE
LLVM and SPIRV-LLVM-Translator pulldown (WW15)

### PR DESCRIPTION
**Pull request by againull**
LLVM: https://github.com/llvm/llvm-project/commit/ab2e7666c20d00a43b958e91c24991c973c81393
SPIRV-LLVM-Translator: https://github.com/KhronosGroup/SPIRV-LLVM-Translator/commit/0ef30a3dfeec433b6d4de4e9cc2a6817032b93e3

### CODE REVIEW

**Comment by againull** [comment]
@sarnex  Could you please take a look at the test in the following commit: https://github.com/intel/llvm/pull/9017/commits/518fb2385be6a7247fea864e4cdbf33551446805

For some reasons test don't pass if it is cherry-picked as is from Khronos repo. It needs modifications. Are these modifications look acceptable?

File: sycl/test-e2e/ESIMD/thread_id_test.cpp [hunk 1]
```diff
@@ -2,7 +2,7 @@
 // UNSUPPORTED: cuda || hip
 // RUN: %clangxx -fsycl %s -o %t.out
 // RUN: %GPU_RUN_PLACEHOLDER %t.out
-// XFAIL: gpu && !esimd_emulator
+// XFAIL: linux && gpu && !esimd_emulator
```
Comment (bader) [inline sycl/test-e2e/ESIMD/thread_id_test.cpp:5]: Please, do not make changes not related to pulling down community changes.

Comment (againull) [reply sycl/test-e2e/ESIMD/thread_id_test.cpp:5]: This test is unexpectedly passing on Windows. Should I ignore that failure?

**Comment by againull** [comment]
> @againull LGTM, probably they are using a different base llvm version which has different llvm.global.annotations structure in this case

Thank you for taking a look!

File: sycl/test-e2e/ESIMD/thread_id_test.cpp (hunk 1 above)
Comment (bader) [reply sycl/test-e2e/ESIMD/thread_id_test.cpp:5]: Yes. The test unexpectedly passing in sycl branch.

Comment (againull) [reply sycl/test-e2e/ESIMD/thread_id_test.cpp]: Reverted my change.

File: llvm-spirv/test/transcoding/registerallocmode.ll [hunk 2]
```diff
@@ -69,4 +76,4 @@ attributes #0 = { noinline nounwind optnone }
 !10 = !{i32 2}
 !11 = !{i32 1}
 !12 = !{i32 0}
-!13 = !{i32 3}
+!13 = !{i32 3}
```
Comment (bader) [inline llvm-spirv/test/transcoding/registerallocmode.ll:79]: Please, revert this change.

**Comment by bader** [comment]
> @againull LGTM, probably they are using a different base llvm version which has different llvm.global.annotations structure in this case

@sarnex, what LLVM version does Khronos repository use?

**Comment by sarnex** [comment]
>@sarnex, what LLVM version does Khronos repository use?

@MrSidims Can you handle this question? I'm not sure why the llvm.global.annotations structure is different in intel/llvm vs khronos. Thx

File: llvm-spirv/test/transcoding/registerallocmode.ll (hunk 2 above)
Comment (againull) [reply llvm-spirv/test/transcoding/registerallocmode.ll]: Reverted.

**Review by bb-sycl — APPROVED** [review]

**Comment by bb-sycl** [comment]
Thu 13 Apr 2023 06:49:02 PM UTC --- Start to merge the commit into sycl branch. It will take several minutes.

**Comment by bb-sycl** [comment]
Thu 13 Apr 2023 06:53:14 PM UTC --- Merge the branch in this PR to base automatically. Will close the PR later.
